ote_sdk/ote_sdk/serialization/label_mapper.py

```diff
@@ -0,0 +1,209 @@
+            "is_empty": instance.is_empty,
+        }
+
+    def backward(self, instance: dict) -> LabelEntity:
+        label_id = IDMapper().backward(instance["_id"])
+
+        domain = instance.get("domain")
+        label_domain = Domain[domain]
+
+        label = LabelEntity(
+            id=label_id,
+            name=instance["name"],
+            color=ColorMapper().backward(instance["color"]),
+            hotkey=instance.get("hotkey", ""),
+            domain=label_domain,
+            creation_date=DatetimeMapper().backward(
+                instance.get("creation_date", None)
+            ),
+            is_empty=instance.get("is_empty", False),
+        )
+        return label
+
+
+class LabelGroupMapper:
+    """
+    This class maps a `LabelGroup` entity to a serialized dictionary, and vice versa
+    """
+
+    def forward(self, instance: LabelGroup) -> dict:
+        return {
+            "_id": IDMapper().forward(instance.id),
+            "name": instance.name,
+            "label_ids": [IDMapper().forward(label.id) for label in instance.labels],
+            "relation_type": instance.group_type.name,
+        }
+
+    def backward(self, instance: dict, all_labels: dict) -> LabelGroup:
+        return LabelGroup(
+            id=IDMapper().backward(instance["_id"]),
+            name=instance["name"],
+            group_type=LabelGroupType[instance["relation_type"]],
+            labels=[
+                all_labels.get(IDMapper().backward(label_id))
+                for label_id in instance["label_ids"]
+            ],
+        )
+
+
+class LabelGraphMapper:
+    """
+    This class maps a `LabelGraph` entity to a serialized dictionary, and vice versa
+    """
+
+    def forward(self, instance: Union[LabelGraph, LabelTree]) -> dict:
+        return {
+            "type": instance.type,
+            "directed": instance.directed,
+            "nodes": [IDMapper().forward(label.id) for label in instance.nodes],
+            "edges": [
+                (IDMapper().forward(edge[0].id), IDMapper().forward(edge[1].id))
+                for edge in instance.edges
+            ],
+        }
+
+    def backward(
+        self, instance: dict, all_labels: dict
+    ) -> Union[LabelTree, LabelGraph]:
+
+        instance_type = instance["type"]
+        if instance_type == "tree":
+            output = LabelTree()
+        elif instance_type == "graph":
+            output = LabelGraph(instance["directed"])
+        else:
+            raise ValueError(f"Unsupported type `{instance_type}` for label graph")
+
+        label_map = {
+            label_id: all_labels.get(IDMapper().backward(label_id))
+            for label_id in instance["nodes"]
+        }
+        for label in label_map.values():
+            output.add_node(label)
+        for edge in instance["edges"]:
+            output.add_edge(label_map[edge[0]], label_map[edge[1]])
+
+        return output
+
+
+class LabelSchemaMapper:
```

Comment:
Is it possible to directly pickle a `LabelSchema`?

Reply:
🤦

Reply:
not relevant anymore